src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -63,6 +65,7 @@ public void drawRoute(final Graphics2D graphics, final RouteDescription routeDes
     final int imageWidth = mapPanel.getImageWidth();
     final int imageHeight = mapPanel.getImageHeight();
     final Dimension imageDimension = new Dimension(imageWidth, imageHeight);
+    screenWidth = mapPanel.getWidth();
```

Comment:
Static state is not a good thing. To understand how this variable is used, you have to know about every call that calls this variable, and then understand the ordering of those calls.
Just pass the 'mapPanel.getWidth()' value to 'getPointOnMap()'

Looking at the source of the data to `getPointOnMap`, it appears we have the feature envy code smell. Looks like getPointOnMap should belong to `mapPanel` and not this class. In that case the only parameter you would need to pass is `routeDescription.getStart()` or `getEnd()`, all of these are derived from `mapPanel` : `xOffset, yOffset, imageDimension), xOffset, yOffset, scale`